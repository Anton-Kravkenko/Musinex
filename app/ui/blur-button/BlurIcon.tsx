import UIcon from '@/ui/icon/defaultIcon/Icon'
import { getHexCode } from '@/utils/getColor'
import { BlurView } from 'expo-blur'
import { useColorScheme } from 'nativewind'
import { FC, PropsWithChildren } from 'react'
import { Pressable } from 'react-native'
import { IblurButton } from './types/IblurButton'

const BlurIcon: FC<PropsWithChildren<IblurButton>> =
	({
		 children,
		 color,
		 iconSize = 21,
		 isSmall = false,
		 icon,
		 ...rest
	 }) => {
		const { colorScheme } = useColorScheme()
		return (
			<Pressable {...!rest.onPress && rest}>
				<BlurView
					intensity={22}
					tint='light'
					className={'items-center justify-center overflow-hidden'}
					style={{
						width: isSmall ? 15 : 40,
						height: isSmall ? 15 : 40,
						borderRadius: isSmall ? 4 : 100
					}}
				>
					{(children || !icon) ? (
						children
					) : (
						<UIcon
							type={'ionicon'}
							onPress={rest.onPress}
							name={icon}
							size={iconSize}
							color={color ? color : colorScheme === 'light' ? getHexCode('dark') : getHexCode('white')}
						/>
					)}
				</BlurView>
			</Pressable>
		)
	}

export default BlurIcon
