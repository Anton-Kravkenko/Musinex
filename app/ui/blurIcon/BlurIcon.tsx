import { BlurButtonProps } from '@/ui/blurIcon/blurButton.types'
import Icon from '@/ui/icon/default-icon/icon'
import { Color } from '@/utils/color'
import { BlurView } from 'expo-blur'
import { FC, memo, PropsWithChildren } from 'react'

const BlurIcon: FC<PropsWithChildren<BlurButtonProps>> = ({
	children,
	color = Color.white,
	iconSize = 21,
	isSmall = false,
	icon = 'alert-outline',
	onPress,
	style,
	...props
}) => (
	<BlurView
		intensity={22}
		tint='light'
		testID='blur-icon'
		className='items-center justify-center overflow-hidden'
		style={[
			{
				width: isSmall ? 15 : 40,
				height: isSmall ? 15 : 40,
				borderRadius: isSmall ? 4 : 8
			},
			style
		]}
		{...props}>
		{children || !icon ? (
			children
		) : (
			<Icon
				onPress={onPress}
				name={icon}
				size={iconSize}
				color={color ? color : Color.white}
			/>
		)}
	</BlurView>
)

export default memo(BlurIcon)
