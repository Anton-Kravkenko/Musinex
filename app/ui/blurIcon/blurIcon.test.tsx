import BlurIcon from '@/ui/blurIcon/BlurIcon'
import {
	cleanup,
	fireEvent,
	render,
	screen
} from '@testing-library/react-native'

afterEach(() => cleanup())
const mockFn = jest.fn()
describe('blurIcon', () => {
	it('press function', () => {
		render(<BlurIcon onPress={mockFn} icon={'search'} />)
		expect(screen.getByTestId('blur-icon')).toBeDefined()
		fireEvent.press(screen.getByTestId('icon-wrapper'))
		expect(mockFn).toHaveBeenCalledTimes(1)
	})
	it('check style', function () {
		const { getByTestId } = render(
			<BlurIcon
				onPress={mockFn}
				icon={'search'}
				style={{
					backgroundColor: 'red'
				}}
			/>
		)
		expect(getByTestId('blur-icon')).toHaveStyle({
			backgroundColor: 'red'
		})
	})
	it('Check className', function () {
		const { getByTestId } = render(
			<BlurIcon onPress={mockFn} icon={'search'} className={'mb-4'} />
		)
		expect(getByTestId('blur-icon')).toHaveProp('className', 'mb-4')
	})
})
