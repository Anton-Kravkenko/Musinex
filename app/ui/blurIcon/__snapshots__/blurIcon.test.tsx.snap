// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blurIcon Render 1`] = `
<View
  className="mb-4"
  style={
    [
      {
        "backgroundColor": "transparent",
      },
      [
        {
          "borderRadius": 8,
          "height": 40,
          "width": 40,
        },
        {
          "backgroundColor": "red",
        },
      ],
    ]
  }
  testID="blur-icon"
>
  <ViewManagerAdapter_ExpoBlurView
    proxiedProperties={
      {
        "intensity": 22,
        "tint": "light",
      }
    }
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="auto"
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderColor": "#1B2021",
          "borderRadius": 16,
          "borderWidth": 0,
          "justifyContent": "center",
          "padding": 8,
        },
        {
          "transform": [
            {
              "scale": 1,
            },
          ],
        },
      ]
    }
    testID="icon-wrapper"
  >
    <Text />
  </View>
</View>
`;
